File: app/scoring/weights.py
HITTER_WEIGHTS = {
    "overall": {
        "split_skill": 0.17,
        "power": 0.13,
        "discipline": 0.11,
        "contact_quality": 0.14,
        "recent_form": 0.12,
        "matchup_fit": 0.12,
        "environment": 0.11,
        "context": 0.10,
    },
    "home_run": {
        "raw_power": 0.22,
        "barrels": 0.16,
        "airball_shape": 0.12,
        "pitcher_damage": 0.16,
        "park_weather": 0.14,
        "matchup_fit": 0.12,
        "recent_form": 0.08,
    },
    "hits": {
        "bat_to_ball": 0.24,
        "on_base": 0.16,
        "split_skill": 0.16,
        "matchup_fit": 0.12,
        "environment": 0.10,
        "recent_form": 0.12,
        "context": 0.10,
    },
    "risk": {
        "strikeouts": 0.34,
        "whiff": 0.18,
        "pitcher_stuff": 0.18,
        "zone_fit": 0.10,
        "context": 0.10,
        "recent_slump": 0.10,
    },
}

PITCHER_WEIGHTS = {
    "overall": {
        "run_prevention": 0.22,
        "strikeouts": 0.16,
        "control": 0.14,
        "contact_suppression": 0.16,
        "recent_form": 0.12,
        "matchup": 0.12,
        "environment": 0.08,
    },
    "strikeout": {
        "strikeouts": 0.34,
        "swing_miss": 0.22,
        "command": 0.10,
        "opponent_k": 0.20,
        "workload": 0.14,
    },
}

File: packages/scoring-engine/src/config/weights.ts
export const hitterWeights = {
  overall: {
    splitSkill: 0.19,
    power: 0.14,
    discipline: 0.1,
    contactQuality: 0.16,
    recentForm: 0.15,
    pitcherVulnerability: 0.14,
    parkBoost: 0.05,
    lineupContext: 0.07,
  },
  homeRun: {
    power: 0.28,
    barrel: 0.24,
    hardHit: 0.16,
    pitcherPowerAllowed: 0.18,
    homeRunParkBoost: 0.14,
  },
  floor: {
    contact: 0.24,
    discipline: 0.18,
    recentForm: 0.18,
    lineupContext: 0.18,
    pitcherVulnerability: 0.12,
    parkBoost: 0.1,
  },
  risk: {
    strikeoutRisk: 0.34,
    lineupVolatility: 0.18,
    weakSplit: 0.2,
    weakForm: 0.16,
    pitcherDifficulty: 0.12,
  },
} as const;

export const pitcherWeights = {
  overall: {
    strikeoutSkill: 0.2,
    control: 0.16,
    contactSuppression: 0.16,
    recentForm: 0.16,
    workload: 0.12,
    matchup: 0.12,
    environment: 0.08,
  },
  strikeoutUpside: {
    strikeoutSkill: 0.36,
    swingMiss: 0.24,
    opponentStrikeouts: 0.24,
    workload: 0.16,
  },
  safety: {
    control: 0.24,
    contactSuppression: 0.24,
    recentForm: 0.18,
    workload: 0.18,
    environment: 0.16,
  },
  blowupRisk: {
    hardHitAllowed: 0.24,
    barrelAllowed: 0.24,
    walkRisk: 0.2,
    opponentPower: 0.18,
    environmentRisk: 0.14,
  },
} as const;
